src/app/events/page.tsx

```diff
@@ -94,23 +108,30 @@ export default function Events({ searchParams }: Props) {
     defaultsTo: 'all-events',
   })
 
-  const { filteredEntries } = useFilter({
+  const { filteredEntries: filteredEventsByLocation } = useFilter({
     searchParams,
     entries: sortedResults,
-    filterKey: CATEGORY_KEY,
-    filterFn: entryMatchesCategoryQuery,
+    filterKey: LOCATION_KEY,
+    filterFn: entryMatchesLocationQuery,
   })
 
-  const { currentPage, pageCount, paginatedResults } = usePagination({
+  const { filteredEntries: filteredEventsByCategory } = useFilter({
     searchParams,
-    entries: filteredEntries,
+    entries: filteredEventsByLocation,
+    filterKey: CATEGORY_KEY,
+    filterFn: entryMatchesCategoryQuery,
   })
```

Comment:
I wonder if we should have something like:

```
// hooks/useFilteredEvents.ts

import { type Event } from '@/types/events'
import { type NextServerSearchParams } from '@/types/searchParams'
import { CATEGORY_KEY, LOCATION_KEY } from '@/constants/searchParams'
import { entryMatchesCategoryQuery, entryMatchesLocationQuery } from '@/utils/filterUtils'
import { useFilter } from '@/hooks/useFilter'

export function useFilteredEvents({
  searchParams,
  entries,
}: {
  searchParams: NextServerSearchParams
  entries: Event[]
}) {
  const { filteredEntries: locationFiltered } = useFilter({
    searchParams,
    entries,
    filterKey: LOCATION_KEY,
    filterFn: entryMatchesLocationQuery,
  })

  const { filteredEntries: filteredEvents } = useFilter({
    searchParams,
    entries: locationFiltered,
    filterKey: CATEGORY_KEY,
    filterFn: entryMatchesCategoryQuery,
  })

  return filteredEvents
}
```

Reply:
We use `filteredEventsByLocation` in

```ts
  const { optionsWithCount: categoryOptionsWithCount } = useListboxOptions({
    collectionName: 'event_entries',
    fieldName: 'category',
    defaultOption: DEFAULT_CATEGORY_FILTER_OPTION,
    entries: filteredEventsByLocation,
  })
```

Hence we have to keep both hooks independent for that reason.

Reply:
Not sure it adds a lot of value. And we end up with the same naming issue as before where `filteredEventsByCategory` includes both filters but its name suggests otherwise.

I would keep it as is, but up to you!